cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java

```diff
@@ -0,0 +1,380 @@
+            // Other - Error occurred, json returned with the error message
+            if (statusCode == HttpURLConnection.HTTP_OK) {
+                outStream.println(org + "/" + name + ":" + version + " pushed to central successfully");
+            } else if (statusCode == HttpURLConnection.HTTP_UNAUTHORIZED) {
+                errStream.println("unauthorized access token for organization: " + org);
+            } else if (statusCode == HttpURLConnection.HTTP_BAD_REQUEST) {
+                try (BufferedReader reader = new BufferedReader(
+                        new InputStreamReader(conn.getErrorStream(), Charset.defaultCharset()))) {
+                    Error errorJsonSchema = new Gson().fromJson(reader, Error.class);
+
+                    if (errorJsonSchema.getMessage() != null && !"".equals(errorJsonSchema.getMessage())) {
+                        throw ErrorUtil.createCommandException(errorJsonSchema.getMessage());
+                    } else {
+                        throw ErrorUtil.createCommandException(
+                                ERR_CANNOT_PUSH + "'" + org + "/" + name + ":" + version + "' reason:" + reader.lines()
+                                        .collect(Collectors.joining("\n")));
+                    }
+                } catch (IOException e) {
+                    throw ErrorUtil.createCommandException(
+                            ERR_CANNOT_PUSH + "'" + org + "/" + name + ":" + version + "' to the remote repository '"
+                                    + conn.getURL() + "'");
+                }
+            } else {
+                throw ErrorUtil.createCommandException(
+                        ERR_CANNOT_PUSH + "'" + org + "/" + name + ":" + version + "' to the remote repository '"
+                                + conn.getURL() + "'");
+            }
+        } finally {
+            conn.disconnect();
+            Authenticator.setDefault(null);
+        }
+    }
+
+    public void pullPackage(String org, String name, String version, Path packagePathInBaloCache,
+            String supportedPlatform, boolean isBuild) {
+        LogFormatter logFormatter = new LogFormatter();
+        if (isBuild) {
+            logFormatter = new BuildLogFormatter();
+        }
+
+        String url = PACKAGES + "/" + org + "/" + name;
+        // append version to url if available
+        if (null != version && !version.isEmpty()) {
+            url = url + "/" + version;
+        }
+
+        initializeSsl();
+        HttpURLConnection conn = createHttpUrlConnection(url);
+        conn.setInstanceFollowRedirects(false);
+        setRequestMethod(conn, Utils.RequestMethod.GET);
+
+        // Set headers
+        conn.setRequestProperty(BALLERINA_PLATFORM, supportedPlatform);
+        conn.setRequestProperty(BAL_LANG_SPEC_VERSION, IMPLEMENTATION_VERSION);
+        conn.setRequestProperty(ACCEPT_ENCODING, IDENTITY);
+        conn.setRequestProperty(USER_AGENT, RepoUtils.getBallerinaVersion());
```

Comment:
We need to pass in the `Accept` header as well.
`'Accept: application/octet-stream'`

Reply:
Added